opencodeblocks/graphics/blocks/codeblock.py

```diff
@@ -111,23 +134,96 @@ def paint(self, painter: QPainter,
+            if not self.resizing_hover:
+                self.resizing_hover = True
+                QApplication.setOverrideCursor(Qt.CursorShape.SizeFDiagCursor)
+        elif self.resizing_hover:
+            self.resizing_hover = False
+            QApplication.restoreOverrideCursor()
+        # Don't call super() because this might override the cursor
+    def hoverLeaveEvent(self, event):
+        """ Triggered when the mouse stops hovering over a block """
+        if self.resizing_hover:
+            self.resizing_hover = False
+            QApplication.restoreOverrideCursor()
+        # Don't call super() because this might override the cursor
+
+    def mousePressEvent(self, event:QGraphicsSceneMouseEvent):
+        """ OCBBlock reaction to a mousePressEvent. """
+        pos = event.pos()
+        resizing_source_code = self._is_in_code_output_resize_area(pos)
+        if (self._is_in_resize_area(pos) or resizing_source_code) and \
+                event.buttons() == Qt.MouseButton.LeftButton:
+            self.resize_start = pos
+            self.resizing = True
+            self.resizing_source_code = resizing_source_code
+            QApplication.setOverrideCursor(Qt.CursorShape.SizeFDiagCursor)
+
+        super().mousePressEvent(event)
+
+    def mouseReleaseEvent(self, event:QGraphicsSceneMouseEvent):
+        """ OCBBlock reaction to a mouseReleaseEvent. """
+        if self.resizing:
+            self.scene().history.checkpoint("Resized block", set_modified=True)
+        self.resizing = False
+        self.resizing_source_code = False
+        QApplication.restoreOverrideCursor()
+        if self.moved:
+            self.moved = False
+            self.scene().history.checkpoint("Moved block", set_modified=True)
+        super().mouseReleaseEvent(event)
+
+    def mouseMoveEvent(self, event:QGraphicsSceneMouseEvent):
+        """ 
+        We override the default resizing behavior as the code part and the display part of the block
+        block can be resized. 
+        """
+        if self.resizing:
+            delta = event.pos() - self.resize_start
+            self.width = max(self.width + delta.x(), self._min_width)
+
+            height_delta = max(delta.y(),
+                # List of all the quantities that must remain negative.
+                # Mainly: min_height - height must be negative for all elements
+                self._min_output_panel_height - self.output_panel_height,
+                self._min_height - self.height,
+                self._min_source_editor_height - self._editor_widget_height()
+            )
+
+            self.height += height_delta
+            if not self.resizing_source_code:
+                self.output_panel_height += height_delta
+
+            self.resize_start = event.pos()
+            self.title_graphics.setTextWidth(self.width - 2 * self.edge_size)
+            self.update()
+        else:
+            super().mouseMoveEvent(event)
+            self.moved = True
```

Comment:
Haven't read everything in details but it's doesn't work well in practice.

Reply:
I don't know what caused this in the first place but I got fixed during the refactor. My guess is that this was due to not calling the `super()` function which prevented PyQt from knowing the size had changed.